pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -0,0 +1,176 @@
+			}
+		}
+
+		deploy.ObjectMeta.Labels = deploymentLabels
+		deploy.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: deploymentLabels,
+			},
+			Spec: corev1.PodSpec{
+				NodeSelector: nodeSelector,
+				InitContainers: []corev1.Container{
+					getCrashDirInitContainer(cephCluster),
+					getCrashChownInitContainer(cephCluster),
+				},
+				Containers: []corev1.Container{
+					getCrashDaemonContainer(cephCluster),
+				},
+				Tolerations:   tolerations,
+				RestartPolicy: corev1.RestartPolicyAlways,
+				HostNetwork:   cephCluster.Spec.Network.IsHost(),
+				Volumes:       append(opspec.DaemonVolumesBase(config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath), ""), keyring.Volume().Admin()),
+			},
+		}
+
+		return nil
+	}
+
+	return controllerutil.CreateOrUpdate(context.TODO(), r.client, deploy, mutateFunc)
+}
+
+func getCrashDirInitContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	crashPostedDir := path.Join(dataPathMap.ContainerCrashDir(), "posted")
+
+	container := corev1.Container{
+		Name: "make-container-crash-dir",
+		Command: []string{
+			"mkdir",
+			"-p",
+		},
+		Args: []string{
+			crashPostedDir,
+		},
+		Image:           cephCluster.Spec.CephVersion.Image,
+		SecurityContext: mon.PodSecurityContext(),
+		Resources:       cephv1.GetCrashCollectorResources(cephCluster.Spec.Resources),
+		VolumeMounts:    opspec.DaemonVolumeMounts(dataPathMap, ""),
+	}
+	return container
+}
+
+func getCrashChownInitContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+
+	return opspec.ChownCephDataDirsInitContainer(
+		*dataPathMap,
+		cephCluster.Spec.CephVersion.Image,
+		opspec.DaemonVolumeMounts(dataPathMap, ""),
+		cephv1.GetCrashCollectorResources(cephCluster.Spec.Resources),
+		&corev1.SecurityContext{},
+	)
+}
+
+func getCrashDaemonContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	cephImage := cephCluster.Spec.CephVersion.Image
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	crashEnvVar := corev1.EnvVar{Name: "CEPH_ARGS", Value: "-m $(ROOK_CEPH_MON_HOST) -k /etc/ceph/admin-keyring-store/keyring -n client.admin"}
+	envVars := append(opspec.DaemonEnvVars(cephImage), crashEnvVar)
+
+	container := corev1.Container{
+		Name: "ceph-crash",
+		Command: []string{
+			"ceph-crash",
+		},
+		Image:        cephImage,
+		Env:          envVars,
+		VolumeMounts: append(opspec.DaemonVolumeMounts(dataPathMap, ""), keyring.VolumeMount().Admin()),
+		Resources:    cephv1.GetCrashCollectorResources(cephCluster.Spec.Resources),
+	}
+
+	return container
+}
+
+func clusterOwnerRef(clusterName, clusterID string) metav1.OwnerReference {
```

Comment:
Can you call the existing `ClusterOwnerRef()` method or what is different?

Reply:
I end up with an import loop if I try :-(
I know it's not ideal...